plugins/bulk_upload/drop_folder_xml/plugins/batch/Engine/KFeedDropFolderEngine.php

```diff
@@ -275,4 +275,41 @@ protected function getMaxFeedBitrate (SimpleXMLElement $feedItem)
 		
 		return max ($bitrates);
 	}
+	
+	/**
+	 * @param string $url
+	 * @return string
+	 */
+	protected function fetchFeedContent ($url)
+	{
+		$user = parse_url ($url, PHP_URL_USER);
+		if (!is_null ($user))
+		{
+			$password = parse_url ($url, PHP_URL_PASS);
```

Comment:
I think it would be much more efficient to call parse_url once with no component argument and read the values from the returned array.